.github/workflows/main.yml

```diff
@@ -0,0 +1,189 @@
+        ports:
+          - 5432:5432
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: pip
+          restore-keys: |
+            pip
+      - name: Install poetry
+        run: |
+          python -m pip install --upgrade pip
+          pip install poetry
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/pypoetry
+          key: poetry-${{ hashFiles('backend/app/poetry.lock') }}
+          restore-keys: |
+            poetry-
+      - name: Install dependencies
+        working-directory: backend/app
+        run: poetry install --no-root
+      - name: Check the code formatting
+        working-directory: backend/app
+        run: |
+          # Activate poetry's virtualenv
+          source `poetry env info --path`/bin/activate
+          # Run formatting tools
+          ./scripts/format.sh
+          # Check that there are no changes
+          git diff --exit-code
+      - name: Lint
+        working-directory: backend/app
+        run: |
+          # Activate poetry's virtualenv
+          source `poetry env info --path`/bin/activate
+          # Run the linter tools
+          ./scripts/lint.sh
+      - name: Run unit tests
+        working-directory: backend/app
+        run: |
+          # Source the .env
+          export $(grep -v '^#' ../../.env | xargs -d '\n')
+          export $(grep -v '^#' .env | xargs -d '\n')
+          # Override the PostgreSQL host name
+          export POSTGRES_SERVER=localhost
+          # Activate poetry's virtualenv
+          source `poetry env info --path`/bin/activate
+          # Run the migration and initial data push
+          ./prestart.sh
+          # Run the tests
+          ./tests-start.sh
+
+  docker-develop-rollout:
+    needs:
+      - test-backend
+      - test-frontend
+    runs-on: k8s-runner
+    if: github.ref == 'refs/heads/develop'
+    strategy:
+      matrix:
+        app: [ "frontend", "backend" ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          version: latest
+      - name: Docker Login Dev
+        uses: docker/login-action@v1
+        with:
+          registry: cridevlab.azurecr.io
+          username: ${{ secrets.DOCKER_DEV_USERNAME }}
+          password: ${{ secrets.DOCKER_DEV_PASSWORD }}
+      - name: Build and push Dev
+        uses: docker/build-push-action@v2
+        with:
+          file: ${{ matrix.app }}/Dockerfile
+          context: .
+          push: true
+          tags: cridevlab.azurecr.io/open-pantheon-${{ matrix.app }}:latest
+          build-args: |
+            BACKEND_URL=https://openpantheon-api.k8s.dev.azure.cri-paris.tech
+            MIXPANEL_PROJECT_TOKEN=595d22a08b4159d437f396461e50e26c
```

Comment:
The build_args should be in the matrix variables since they only apply to the frontend

Reply:
build args dependant of the matrix is a bit tricky to implement. I now have dedicated steps for frontend and backend.